### PR TITLE
IndexError: list index out of range #17

### PR DESCRIPTION
**Pull request by stormcenter**
Traceback (most recent call last):
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/gradio/queueing.py", line 388, in call_prediction
    output = await route_utils.call_process_api(
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/gradio/route_utils.py", line 219, in call_process_api
    output = await app.get_blocks().process_api(
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/gradio/blocks.py", line 1437, in process_api
    result = await self.call_function(
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/gradio/blocks.py", line 1109, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/gradio/utils.py", line 650, in wrapper
    response = f(*args, **kwargs)
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "webUI.py", line 245, in process
    keypath = process2(*args_wo_process3)
  File "/home/chi/anaconda3/envs/rerender/lib/python3.8/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "webUI.py", line 402, in process2
    frame = cv2.imread(imgs[i + 1])
IndexError: list index out of range

I have meet this issue ,and my video total frame number is 282, the imgs number is 262, the whole key frames is 28, so I think maybe there are some issues on keyframe caculator

### CODE REVIEW

File: webUI.py [hunk 1]
```diff
@@ -397,10 +397,14 @@ def process2(*args):
     pre_result = first_result
     pre_img = first_img
 
-    for i in range(0, cfg.frame_count - 1, cfg.interval):
+    for i in range(0, len(imgs) - 1, cfg.interval):
         cid = i + 1
-        frame = cv2.imread(imgs[i + 1])
         print(cid)
+        print(len(imgs))
```
Comment (SingleZombie) [inline webUI.py:403]: I suggest to remove this line.

File: webUI.py [hunk 2]
```diff
@@ -397,10 +397,14 @@ def process2(*args):
...
+        if cid <= (len(imgs) - 1):
+            frame = cv2.imread(imgs[i + 1])
```
Comment (SingleZombie) [inline webUI.py:405]: It's better to use `cid` rather than `i+1`.

File: webUI.py [hunk 3]
```diff
@@ -397,10 +397,14 @@ def process2(*args):
     pre_result = first_result
     pre_img = first_img
 
-    for i in range(0, cfg.frame_count - 1, cfg.interval):
+    for i in range(0, len(imgs) - 1, cfg.interval):
```
Comment (SingleZombie) [inline webUI.py:400]: The end of the loop should be `min(len(imgs), cfg.frame_count) - 1`.

**Review by SingleZombie — CHANGES_REQUESTED** [review]
Thanks for contribution. Would you like to also update the same codes in `rerender.py?`
https://github.com/williamyang1991/Rerender_A_Video/blob/619e88febda844a6df2f536354e1c619c2416dc7/rerender.py#L216

**Comment by stormcenter** [comment]
done

**Review by SingleZombie — APPROVED** [review]
LGTM

**Pull request merged**